core/src/player.rs

```diff
@@ -377,19 +377,14 @@ impl Player {
                 .stage
                 .set_movie(context.gc_context, context.swf.clone());
 
-            let global_domain = context.avm2.global_domain();
-            let mut global_activation =
-                Avm2Activation::from_domain(context.reborrow(), global_domain);
-            let domain = Avm2Domain::movie_domain(&mut global_activation, global_domain);
-
-            let mut activation =
-                Avm2Activation::from_domain(global_activation.context.reborrow(), domain);
+            let stage_domain = context.avm2.stage_domain();
+            let mut activation = Avm2Activation::from_domain(context.reborrow(), stage_domain);
 
             activation
                 .context
                 .library
                 .library_for_movie_mut(activation.context.swf.clone())
-                .set_avm2_domain(domain);
+                .set_avm2_domain(stage_domain);
```

Comment:
this sounds slightly weird. In the scenario where we overwrite the root movie:
- the SWF loads with its own domain
- while the SWF loads, it sets all its classes into its domain
- when we make this SWF's root movie into the stage root movie, we overwrite its domain with stage domain
- ergo, all the definitions of the child movie are forgotten?
That doesn't sound right.

Are you sure that it's not the other way around - the child movie's domain becoming the new stage domain (and the old stage domain remaining as accessible parent)? I don't know if that's what FP does, but it sounds like this way less content would break.

Reply:
We never delete things from a domain, or overwrite the overall domain. The linkage is only from a movie to a domain, so replacing the root movie will continue to use the existing stage domain (and all of the loaded classes will stay loaded).